src/Service/Resolvers/PostgresQueryBuilder.cs

```diff
@@ -110,7 +110,7 @@ public string Build(SqlUpsertQueryStructure structure)
         {
             // https://stackoverflow.com/questions/42668720/check-if-postgres-query-inserted-or-updated-via-upsert
             // relying on xmax to detect insert vs update breaks for views
-            string predicates = JoinPredicateStrings(Build(structure.Predicates), structure.DbPolicyPredicates);
+            string predicates = JoinPredicateStrings(Build(structure.Predicates), structure.DbPolicyPredicatesForOperation[Config.Operation.Update]);
```

Comment:
Shouldn't this be Upsert?

Reply:
No, upsert is a compound operation. The `predicates` here is used only for update portion of the upsert query.